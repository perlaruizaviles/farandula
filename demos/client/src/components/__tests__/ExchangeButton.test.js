import React from 'react';
import {shallow} from 'enzyme';
import ExchangeButton from '../Common/ExchangeButton';

describe('Rendering ExchangeButton ', () => {

  const props = {
    handleExchange: () => {}
  };

  const tree = shallow(<ExchangeButton {...props} />);

  it('Should create an snapshot for ExchangeButton', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Renders Button', () => {
    expect(tree.find('Button').length).toBe(1);
  });

  it('Renders Correct Icon', () => {
    expect(tree.find('Icon').length).toBe(1);
  });

  it('Renders Correct exchange Icon', () => {
    expect(tree.find('Icon').props().name).toBe('exchange');
  });
});