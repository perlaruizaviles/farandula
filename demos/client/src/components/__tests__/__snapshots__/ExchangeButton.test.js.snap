exports[`Rendering ExchangeButton  Should create an snapshot for ExchangeButton 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    as="button"
    icon={true}
    onClick={[Function]}>
    <Icon
        as="i"
        name="exchange" />
</Button>,
  "nodes": Array [
    <Button
      as="button"
      icon={true}
      onClick={[Function]}>
      <Icon
            as="i"
            name="exchange" />
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ExchangeButton
        handleExchange={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleExchange": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          as="button"
          icon={true}
          onClick={[Function]}>
          <Icon
                    as="i"
                    name="exchange" />
</Button>,
        "_debugID": 2,
        "_renderedOutput": <Button
          as="button"
          icon={true}
          onClick={[Function]}>
          <Icon
                    as="i"
                    name="exchange" />
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ExchangeButton
    handleExchange={[Function]} />,
}
`;
